### DISCUSSION TITLE
Illustrative samples missing on how to use Valtio?

### DISCUSSION THREAD

**Discussion by sankarkumar23** [pmndrs/valtio#779] [General]
Valtio offers impressive advantages by eliminating the redundant/boiler pate code that Redux introduced. However, I'm struggling to find comprehensive real-world examples that seamlessly guide developers towards adopting Valtio. For instance, it would be immensely beneficial to have practical samples showcasing, for example, how to manage authentication, authorization info, current user information, or Material UI theme using Valtio. Such illustrative examples could significantly accelerate the learning curve and encourage wider adoption of Valtio within the development community.

**Comment by overthemike** [comment]
I love the idea of creating good examples to showcase both valtio's benefits and even some of it's gotchas. I have a few questions on some of the samples you asked about:

> Manage Authentication

Can you help by specifying what you're thinking might be confusing or what kind of example you'd think would be helpful? Authentication isn't something valtio handles, but is rather _used_ by whatever authentication strategy you decide to use. Here's a video of using valtio alongside firebase for auth that might help: https://www.youtube.com/watch?v=hUi5RcRyhxM

> Authorization Info

This is another piece that can sometimes mean many things. Are you referring to stuff like permissions? Valtio can store those permissions but I don't think that it should be doing any authorization logic per say...

> Current User Information

This seems like a pretty normal piece of state like any other, right? Do you think it might help people click with the concepts better by using this kind of thing instead of something like a todo app perhaps?

> Material UI Theme

Can you explain what you mean? Material UI, I believe, uses a it's own theme provider wrapper component. What would valtio really have to do here?

I don't mean to knock down your suggestions at all so I hope this isn't coming across as such. I love the idea of getting content created that would bring more visibility and understanding to the library in whatever way that would be helpful and draw more attention.

Is there a practical example you have in mind you'd like to see as a tutorial or something similar?

**Comment by dai-shi** [comment]
`./docs` are maintained by the community. Suggestions and contributions are welcome.

**Comment by pooya-badiee** [comment]
Would an examples/showcase section help address this?
Maybe something like:

https://redux.js.org/introduction/examples
https://jotai.org/docs/basics/showcase

I personally don't think examples like Material UI are ideal, since most of the logic won't contribute to learning Valtio specifically.

**Reply by overthemike** [reply]
> Would an examples/showcase section help address this?

I think this would be awesome. Could start off similar with how jotai does it. It would be a good amount of work, but it doesn't need to be done all at once.

> I personally don't think examples like Material UI are ideal, since most of the logic won't contribute to learning Valtio specifically.

Agreed - this wouldn't really necessarily belong in the docs, but if someone creates a tutorial or post that may use other libs, I wouldn't want to discourage it either.d

**Reply by pooya-badiee** [reply]
Maybe a good way to handle this so examples can be code reviewed and discussed without creating overly large PRs:
We could create an examples folder. Once we have a reasonable amount of content, we can publish a page for it in the docs.
Alternatively, there could be a separate branch where we submit and review examples one by one, then push them all together at the end.

what do you suggest?

**Reply by overthemike** [reply]
It would be a fine way to start. Having code available somewhere like stackblitz though might make it easier for people to play around with examples, but that can be done later too. I'd say a good start would be making a list of helpful examples so we have some clear ideas.

As far as how to approach it, I think just sticking with forks/PRs is probably a better idea than making a separate branch. People could also create their own repos with examples as a good way to start.
